cirq-core/cirq/sim/classical_simulator.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+from collections import defaultdict
+from cirq.sim.simulator import SimulatesSamples
+from cirq import ops, protocols
+from cirq.study.resolver import ParamResolver
+from cirq.circuits.circuit import AbstractCircuit
+from cirq.ops.raw_types import Qid
+import numpy as np
+
+
+class ClassicalStateSimulator(SimulatesSamples):
+    """basic simulator that only accepts cirq.X, cirq.ISwap, and cirq.CNOT gates
+
+    Run a simulation, mimicking quantum hardware.
+
+      Args:
```

Comment:
Let's also point out that the simulator does not handle symbols yet but they may be supported in the future.